### DISCUSSION TITLE
ebus topology

### DISCUSSION THREAD

**Discussion by cociweb** [john30/ebusd#628] [Q&A]
Hello,
I'm completelly new with ebus, and I tried to gather some info on the net, but google gives me nowadays electronic buses, instead of relevant vaillant infos. :)

So, I have a couple(/tons?) of basic questions:
1) in connection with wiring: 
I have an EcoTec heater with ebus connector <+++ wire ++++> a VRC700 thermostat with ebus connector. How should I wire the things to add the 'ebus adapter v3' and keep the VRC700 thermostat working as well? (star topology?) Simply sweep the 2*3 (+/-) wires in one junction?
Or I'm wrong, and I need to make a loop between them? (In case of loop, does the order matter?)
Or I'm totally  wrong, and I need an external coupler/divider? (I'm also lost in the VR products.)
Or I'm extremelly totally wrong and it is not possible the add the 'ebus adaptor V3' AND keep the VRC700 thermostat for manual overwrite/ adjustment purposes.
In my setup what kind of config file should be used when I comission the 'ebus adaptor v3'?

2) in connection with adressing: 
Is ebus is a standard bi-directional bus and not a P2P connection? Does it requires addressing? If it requires, I assume that the heater and the thermostat has an own hardcoded address. Does the 'ebus adapter V3' requires unique address? what/how should I set? or the address is coming from the id of the onboard PIC?

3) in connection with additional hardware elements:
In the wiki/docs I did not found explicit answer: Does the 'ebus adapter V3' requires must have additional hardware (such as an RPI)? - So it's an adapter for ebus and not an adapter for RPI (RPI connectors are just for 'fun' and the onboard PIC do 'everything'.) .
The docs says the adapter should be powered via the wifi module (if it's packed with the serial-wifi gateway) but there is no recommendation. what should be the Volts and Amper ranges - ? 

4) "The current batch is already fully reserved (requests before Jul. 24th, 2022)" Do you have any info what is the target demand for the new batch? What is the calendar estimation for the fulfillment of the next batch?  - my expectation is that in the winter nobody wants to upgrade his/her ebus. :)

(PS: Sorry If I placed my questions in one discussion and also in the ebusd Q&A topic, but ebus-esp has no discussion platform+the ebusd.eu redirected me to here.)

Thank you for any supportive answer.

**Comment by cociweb** [comment]
Any answers for the above mentioned questions would be appreciated!

**Comment by DiaMontPhil** [comment]
@cociweb
1.+2. I recognize most of your questions as those were mine too some time ago until I read : https://github.com/john30/ebusd/wiki/eBUS-background
3. The v3 adapter converts the ebus signals to RS232. This RS232 signal needs to be read by a controller for example a raspberry pi (with it's GPIO). In case of a Wifi module " LOLIN/Wemos D1 mini with ESP-8266 ". The power is supplied directly via the USB connector of the Wemos. A quick look to the specifications of the Wemos: max voltage 6V.  Power Consumption @ 5V around 70mA.
Good luck integrating the eBus in your system! :-)

**Comment by john30** [comment]
1+2 should be answered with the link already. anyway, each eBUS device needs its own address and as long as you don't send anything to the bus, ebusd does not need its own (then it is just listening to the traffic). it is a lot more fun with sending though (as actively reading a register from a device also requires sending symbols to the bus).
3. ebusd (or some other tool to interpret the eBUS traffic) needs to run somewhere. it is way too big to fit into the PIC, so that's why it usually runs on a Linux host. the PIC is only for doing very low level and time critical bus access ("arbitration" in eBUS wording). if using the RPI variant of the adapter, it will be directly powered by RPI. the same applies to WIFI variant which needs some USB power (just around 50mA more than the ESP needs itself).
4. especially during autumn/winter people are active in eBUS :) unfortunately there is no way to give estimate to the next batch delivery as the awful situation with the chips is still ongoing and no end in sight
